ocaml/xapi/xapi_cluster.ml

```diff
@@ -71,7 +71,11 @@ let create ~__context ~network ~cluster_stack ~pool_auto_join ~token_timeout ~to
         set_ha_cluster_stack ~__context;
         cluster_ref
       | Result.Error error ->
```

Comment:
I think if we are in this match case, the refs aren't in the DB yet, because they are put into the DB in case of `Result.Ok`. So we don't have to clean them up.